yt_dlp/extractor/rokfin.py

```diff
@@ -0,0 +1,740 @@
+            yielded_result_counter = 0  # How many search results have been yielded?
+            POST_DATA = {'query': query, 'page': {'size': ENTRIES_PER_PAGE}, 'facets': {'content_type': {'type': 'value', 'size': ENTRIES_PER_PAGE}, 'creator_name': {'type': 'value', 'size': ENTRIES_PER_PAGE}, 'premium_plan': {'type': 'value', 'size': ENTRIES_PER_PAGE}}, 'result_fields': {'creator_twitter': {'raw': {}, 'snippet': {'size': ENTRIES_PER_PAGE, 'fallback': True}}, 'content_id': {'raw': {}, 'snippet': {'size': ENTRIES_PER_PAGE, 'fallback': True}}, 'creator_username': {'raw': {}, 'snippet': {'size': ENTRIES_PER_PAGE, 'fallback': True}}, 'creator_instagram': {'raw': {}, 'snippet': {'size': ENTRIES_PER_PAGE, 'fallback': True}}, 'post_comments': {'raw': {}, 'snippet': {'size': ENTRIES_PER_PAGE, 'fallback': True}}, 'post_text': {'raw': {}, 'snippet': {'size': ENTRIES_PER_PAGE, 'fallback': True}}, 'content_description': {'raw': {}, 'snippet': {'size': ENTRIES_PER_PAGE, 'fallback': True}}, 'content_title': {'raw': {}, 'snippet': {'size': ENTRIES_PER_PAGE, 'fallback': True}}, 'post_updated_at': {'raw': {}, 'snippet': {'size': ENTRIES_PER_PAGE, 'fallback': True}}, 'creator_youtube': {'raw': {}, 'snippet': {'size': ENTRIES_PER_PAGE, 'fallback': True}}, 'content_type': {'raw': {}, 'snippet': {'size': ENTRIES_PER_PAGE, 'fallback': True}}, 'creator_name': {'raw': {}, 'snippet': {'size': ENTRIES_PER_PAGE, 'fallback': True}}, 'creator_facebook': {'raw': {}, 'snippet': {'size': ENTRIES_PER_PAGE, 'fallback': True}}, 'id': {'raw': {}, 'snippet': {'size': ENTRIES_PER_PAGE, 'fallback': True}}, 'premium_plan': {'raw': {}, 'snippet': {'size': ENTRIES_PER_PAGE, 'fallback': True}}}}
+
+            for page_n in itertools.count(1) if n_results == float('inf') else range(1, max_pages_to_download + 1):
+                POST_DATA['page']['current'] = page_n
+
+                if RokfinSearchIE.service_url and RokfinSearchIE.service_access_key:
+                    # Access has already been established.
+                    srch_res = self._download_json(
+                        RokfinSearchIE.service_url, self._SEARCH_KEY, headers={'authorization': RokfinSearchIE.service_access_key},
+                        data=json.dumps(POST_DATA).encode('utf-8'), encoding='utf-8',
+                        note=f'Downloading search results (page {page_n}' + (f' of {min(pages_total, max_pages_to_download)}' if pages_total is not None and max_pages_to_download is not None else '') + ')',
+                        fatal=True)
+                else:
+                    self.write_debug(msg='gaining access')
+
+                    # Try all possible combinations between service_urls and service_access_keys and see which one works.
+                    # This should succeed on the first attempt, but no one knows for sure.
+                    for service_url, service_access_key in (lambda p: itertools.product(p[0], p[1]))(self._get_access_credentials()):
```

Comment:
Have you ever encountered a case where this doesnt work with the first attempt? If not, these seems like over-engineering. The whole access token code can be moved to `_real_initialize` instead